Documentation/DesignDocs/corewcfqueue-design.md

```diff
@@ -0,0 +1,308 @@
+- using the `QueueBaseTransportBindingElement` which is fetched from above step, `BuildQueueTransportPump` method is called to get the `QueueTransportPump`.
+ - In the end, for every service dispatcher, a `QueueTransportContext` object is created which stores objects of type `QueueTransportPump`, `ServiceDispatcher`, `QueueBaseTransportBindingElement`,`MessageEncoderFactory` and `QueueHandShakeDelegate`.
+- Since `QueuePollingService` implements `IHostingService`, it implements the `StartAsync` and `StopAsync` methods.
+- `StartAsync` is called by WebHost and it performs the below task.
+```csharp
+public async Task StartAsync(CancellationToken cancellationToken)
+{
+    var tasks = _queueTransportContexts.Select(_queueTransport => StartFetchingMessage(_queueTransport));
+    await Task.WhenAll(tasks);
+}
+
+private async Task StartFetchingMessage(QueueTransportContext queueTransport)
+{
+    await queueTransport.QueuePump.StartPumpAsync(queueTransport, CancellationToken.None);
+}
+
+```
+- For StartAsync
+ - First we create a list of tasks`StartFetchingMessage` for each `QueueTransportContext` (populated during Init).
+ - `QueueTransportPump` from `QueueTransportContext` is used by each `StartFetchingMessage` to call the `StartPumpAsync` method.
+ - If the `QueueTransportPump` is based on a push message mechanism, it awaits until a message is received and invokes service as soon as the message is received.
+ - If the `QueueTransportPump` is using `DefaultQueueTransportPump`, it creates multiple concurrent tasks to invoke `FetchAndProcessAsync` method based on concurrency level.
+ -The `FetchAndProcessAsync` method of QueueTransport (concrete transport) calls `ReceiveQueueMessageContextAsync` to retrieve the message and pass the handle to the middleware for processing. 
+- StopAsync cancels all tasks and, as a result, the fetching of messages is stopped.
+#### QueueFetchMessage
+- `QueueFetchMessage` is the first middleware which is invoked as soon as a message is fetched from the queue.
+- The `InvokeAsync` method reads messages from pipereader and encodes them into a concrete `Message` objet and puts them back to `QueueMessageContext` and invokes the next delegate , which is `QueueProcessMessage`.
+#### QueueProcessMessage
+- Invokes the `DispatchAsync` method of the channel dispatcher by passing `QueueMessageContext` object and ending the middleware pipeline.
+- As a part of `DispatchAsync`, the service operation is called, processing of messages is done, eventually `QueueMessageContext` (inherits `RequestContext`) `ReplyAsync` method is called.
+- The `ReplyAsync` mechanism is used to inform the concrete transport that a message processing has finished with a status of success or failure by invoking an Action `DispatchResultHandler`. 
+- By calling the `DispatchResultHandler` each queue providers which support timeouts or requires notification to prevent reprocessing a message can take appropriate actions. 
```

Comment:
inform the transport implementation that message processing has finished with a status of success of failure. It signals the status result by invoking a `DispatchResultHandler` delegate.